pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -27,40 +27,95 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/extension/extensiontest"
+
 	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
 )
 
-func TestNewOpampAgent(t *testing.T) {
-	cfg := createDefaultConfig()
+const errMsgRemoteConfigNotAccepted = "OpAMP agent does not accept remote configuration"
+const errMsgInvalidConfigName = "cannot validate config named " +
+	"service::pipelines::logs/localfilesource/0aa79379-c764-4d3d-9d66-03f6df029a07: " +
+	"references processor \"batch\" which is not configured"
+
+func defaultSetup() (*Config, extension.Settings) {
+	cfg := createDefaultConfig().(*Config)
 	set := extensiontest.NewNopSettings()
-	set.BuildInfo = component.BuildInfo{Version: "test version", Command: "otelcoltest"}
-	o, err := newOpampAgent(cfg.(*Config), set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-	assert.Equal(t, "otelcoltest", o.agentType)
-	assert.Equal(t, "test version", o.agentVersion)
-	assert.NotEmpty(t, o.instanceId.String())
-	assert.Empty(t, o.effectiveConfig)
-	assert.Nil(t, o.agentDescription)
+	return cfg, set
 }
 
-func TestNewOpampAgentAttributes(t *testing.T) {
-	cfg := createDefaultConfig()
-	set := extensiontest.NewNopSettings()
-	set.BuildInfo = component.BuildInfo{Version: "test version", Command: "otelcoltest"}
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceName, "otelcol-sumo")
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceVersion, "sumo.0")
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceInstanceID, "f8999bc1-4c9b-4619-9bae-7f009d2411ec")
-	o, err := newOpampAgent(cfg.(*Config), set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-	assert.Equal(t, "otelcol-sumo", o.agentType)
-	assert.Equal(t, "sumo.0", o.agentVersion)
-	assert.Equal(t, "7RK6DW2K4V8RCSQBKZ02EJ84FC", o.instanceId.String())
+func setupWithRemoteConfig(t *testing.T, d string) (*Config, extension.Settings) {
+	cfg, set := defaultSetup()
+	cfg.RemoteConfigurationDirectory = d
+	return cfg, set
+}
+
+func setupWithBuildInfo(version, command string) (*Config, extension.Settings) {
```

Comment:
what is the point of this function? can't this line be in ```defaultSetup```?

Reply:
this was being used in the test ```setupWithBuildInfo("test version", "otelcoltest")```, shouldn't this be in the setup function?

Reply:
some functions use `	set.BuildInfo = component.BuildInfo{Version: version, Command: command}`
while some do not, so in such places we use either defaultSetup or setupWithBuilderInfo depending upon the testcase

Reply:
i agree with that, then you can use a blank identifier to make it cleaner. so that way instead of writing smaller functions, it can all be in one function instead.